cmake/ActsCompilerOptions.cmake

```diff
@@ -21,32 +26,19 @@ if(DEFINED CMAKE_CXX_STANDARD)
   endif()
 endif()
 
-# This adds some useful conversion checks like float-to-bool, float-to-int, etc.
-# However, at the moment this is only added to clang builds, since GCC's -Wfloat-conversion 
-# is much more aggressive and also triggers on e.g., double-to-float
-if(CMAKE_CXX_COMPILER_ID MATCHES "Clang|AppleClang") 
- set(ACTS_CXX_FLAGS "${ACTS_CXX_FLAGS} -Wfloat-conversion")
-endif()
 
 if(ACTS_ENABLE_CPU_PROFILING OR ACTS_ENABLE_MEMORY_PROFILING)
   message(STATUS "Added -g compile flag")
-  set(ACTS_CXX_FLAGS "${ACTS_CXX_FLAGS} -g")
+  list(APPEND cxx_flags "-g")
 endif()
 
-# Acts linker flags
-set(ACTS_EXE_LINKER_FLAGS_DEBUG "--coverage")
-set(ACTS_SHARED_LINKER_FLAGS_DEBUG "--coverage ")
+string(REPLACE " " ";" explicit_flags "${CMAKE_CXX_FLAGS}")
+list(APPEND cxx_flags "${explicit_flags}")
+list(REMOVE_DUPLICATES cxx_flags)
```

Comment:
And then we ignore duplicates? Probably doesn't break anything, so might be ok.

Reply:
:thinking: On first order, yes... I think it will anyway be very rare that this would come up. But even when it does, the compiler frontends don't tend to mind duplicate flags all too much.

At the same time, a really messed up set of flags like `"-std=c++20 -std=c++17 -std=c++20"` would result in a different outcome with the duplicates removed. :thinking: I think it's because of all of these possibilities that the CMake core code doesn't try to be too smart with this.

Reply:
Ok, switched back to just string composition now.

Reply:
Can you check again, and hit approve if you don't have further suggestions?